src/ts/ConvivaAnalytics.ts

```diff
@@ -631,6 +631,9 @@ export class ConvivaAnalytics {
 
     this.lastSeenBitrate = null;
     this.convivaVideoAnalytics.reportPlaybackMetric(Conviva.Constants.Playback.BITRATE, bitrateKbps);
+    if (event.targetQuality.frameRate) {
+      this.convivaVideoAnalytics.reportPlaybackMetric(Conviva.Constants.Playback.RENDERED_FRAMERATE, event.targetQuality.frameRate);
+    }
```

Comment:
As this is called inside the `onVideoQualityChanged` it could happen that `event.targetQuality.frameRate` changed from the previous quality.

When we just check `if (event.targetQuality.frameRate)` but the `frameRate` changes to `undefined / null` we would still report the previous framerate 🤔 Is there a way to unset the `Conviva.Constants.Playback.RENDERED_FRAMERATE` if it's not set?

Reply:
I do not see a way to _unset_ this metric. The intention is for this metric to change and not report it if we see `undefined / null`.
What is the situation where we would report a previous framerate?

Reply:
I mean like different qualities could have different / no framerates e.g. (don't know 100%) switching to an audio-only quality could mean that there maybe is no framerate present 🤔 

@dweinber maybe has more insights into how the web-player behaves here

Reply:
It's my understanding that if the qualities have different framerate that should be reported to calculate the average FPS over a session.
I'm unsure on how no framerate affects the end calculations in conviva and don't see anything in their documentation for handling this case.

Reply:
Ok 🤔 Will approve the PR 🙂

Reply:
Switching between audio/video and audio-only renditions is anyway not supported in the web player @stonko1994 (except Safari) 😅 

Framerate is optional, so theoretically, one rendition/Representation could have it set while another has not. However, I can't remember having seen this in practice, it's usually all or nothing. So should be fine.

Reply:
Ok, thx for clarifying 🙂